dubbo-registry/dubbo-registry-nacos/src/main/java/org/apache/dubbo/registry/nacos/NacosRegistry.java

```diff
@@ -323,7 +325,7 @@ private Set<String> getServiceNames0(URL url) {
 
         final Set<String> serviceNames;
 
-        if (serviceName.isConcrete()) { // is the concrete service name
+        if (supportLegacyServiceName && serviceName.isConcrete()) { // is the concrete service name
```

Comment:
Will Dubbo still subscribe to the original service name when `supportLegacyServiceName` is set to `false`? Because I didn't see the place where the original serviceName is added into serviceNames.

Reply:
In line 338

Reply:
fixed